### PR TITLE
Removed old networking code, Re-add hostname resolution, Fix minimeso…

### PR DESCRIPTION
**Pull request by philwinder**
…s system tests.

### CODE REVIEW

**Comment by philwinder** [comment]
System tests passes on https://ci.infra.container-solutions.com/view/All/job/ElasticsearchPullRequests/202/console

Tested on Mesos infra. Note how hostnames were being resolved to slave's { 172.31.17.16, 172.31.17.17} - private IP's.

screen shot 2015-11-03 at 09 33 15

File: scheduler/src/main/java/org/apache/mesos/elasticsearch/scheduler/controllers/TasksController.java [hunk 1]
```diff
@@ -47,7 +47,11 @@ private GetTasksResponse from(Map.Entry<String, Task> task) {
     }
 
     private String toFormattedAddress(InetSocketAddress clientAddress) {
-        return String.format("%s:%s", clientAddress.getAddress().getHostAddress(), clientAddress.getPort());
+        if (clientAddress.isUnresolved()) { // Protect against unresolved IP addresses.
+            return "UNKNOWN";
```
Comment (mwl) [inline scheduler/src/main/java/org/apache/mesos/elasticsearch/scheduler/controllers/TasksController.java]: Isn't this a bit dangerous? If we're in a network without dns `"Unknown"` will be all over the place?

File: system-test/src/main/java/org/apache/mesos/elasticsearch/systemtest/Main.java [hunk 2]
```diff
@@ -22,7 +20,7 @@ public static void main(String[] args) throws InterruptedException {
         MesosCluster cluster = new MesosCluster(
             MesosClusterConfig.builder()
                 .mesosImageTag(MESOS_IMAGE_TAG)
-                .slaveResources(new String[]{"ports(*):[9200-9200,9300-9300]", "ports(*):[9201-9201,9301-9301]", "ports(*):[9202-9202,9302-9302]"})
+                .slaveResources(new String[]{"ports(*):[4000-4000,5000-5000]", "ports(*):[4001-4001,5001-5001]", "ports(*):[4002-4002,5002-5002]"})
```
Comment (mwl) [inline system-test/src/main/java/org/apache/mesos/elasticsearch/systemtest/Main.java]: Just out of curiosity: Why the new ports?

**Comment by philwinder** [comment]
Todo: Do hostname resolve check in offers check.

File: system-test/src/main/java/org/apache/mesos/elasticsearch/systemtest/Main.java (hunk 2 above)
Comment (philwinder) [inline system-test/src/main/java/org/apache/mesos/elasticsearch/systemtest/Main.java]: I will revert back. No reason. I was just testing that the docker ports expose worked with different ports and forgot to move them back.

File: scheduler/src/main/java/org/apache/mesos/elasticsearch/scheduler/controllers/TasksController.java (hunk 1 above)
Comment (philwinder) [inline scheduler/src/main/java/org/apache/mesos/elasticsearch/scheduler/controllers/TasksController.java]: Yes. After the resolveable hostname offer check, it should never reach this line. Thanks.

**Comment by mwl** [comment]
LFTM

**Pull request merged**